p_sets/3/generate.rb

```diff
@@ -4,23 +4,19 @@ class RandomArrayGenerator
 
   def initialize(length, seed = nil)
     @length = length
-    @seed = seed
+    srand(seed) if seed
   end
 
   def array
     Array.new(length) { (rand * EXPONENT).ceil }
   end
 
-  def seed_rand
-    srand seed
-  end
 end
 
-if ARGV.count > 2 || ARGV.empty?
-  exit 1
+if ARGV.count > 2 || ARGV.count < 1
```

Comment:
In the instructions it specified:

```Usage: generate n [s]
As this output suggests, this program expects one or two command-line arguments. The first, n, is required; it indicates how many pseudorandom numbers you’d like to generate. The second, s, is optional, as the brackets are meant to imply; if supplied, it represents the value that the pseudorandom-number generator should use as its "seed."  ```

Sorry, I should have pasted that into the instructions doc.